hgtv/views/channel.py

```diff
@@ -128,11 +140,12 @@ def playlist_new_modal(channel, video):
 
 @app.route('/<channel>/new/stream', methods=['GET', 'POST'])
 @lastuser.requires_login
+@render_with({'text/html': 'index.html.jinja2', 'application/json': add_new_video})
 @load_models(
     (Channel, {'name': 'channel'}, 'channel'),
     permission='new-video')
 def stream_new_video(channel):
     """
     Add a new video to stream playlist
     """
-    return add_new_video(channel, playlist=None)
+    return dict(channel=channel, playlist=None)
```

Comment:
Since channel is discoverable as `playlist.channel`, here too `playlist.current_access()` should suffice (except we currently can't wrap relationships in RoleMixin, as that is pending).